File: code_combat/7_Sarven_Desert/329-Wandering_Souls/wandering_souls.py
while True:
    enemies = hero.findEnemies()
    for enemy in enemies:
        while enemy.health > 0:
            hero.attack(enemy)
    
    items = hero.findItems()
    for item in items:
        while hero.distanceTo(item) > 2:
            hero.moveXY(item.pos.x, item.pos.y)

File: projects/code_combat/7_Sarven_Desert/329-Wandering_Souls/wanderingSouls.js
// Defeat skeletons and collect lightstones.

while (true) {
    var enemies = hero.findEnemies();
    var enemyIndex = 0;
    while (enemyIndex < enemies.length) {
        var enemy = enemies[enemyIndex];
        // Attack while enemy has health.
        while (enemy.health > 0) {
            hero.attack(enemy);
        }
        enemyIndex += 1;
    }
    var items = hero.findItems();
    var itemIndex = 0;
    // Iterate over all items.
    while (itemIndex < items.length) {
        var item = items[itemIndex];
        // While the distance greater than 2:
        while (hero.distanceTo(item) > 2)
            // Try to take the item.
            hero.moveXY(item.pos.x, item.pos.y);
        // Don't forget to increase itemIndex.
        itemIndex += 1;
    }
}
